### PR TITLE
Create Lessons

### PR DESCRIPTION
**Pull request by nguyenduyta**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -15,3 +15,24 @@
 //= require jquery_ujs
 //= require turbolinks
 //= require_tree .
+
+function next_word(i) {
+    var result = $('.result');
```
Comment (dieunb) [inline app/assets/javascripts/application.js]: indent

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -13,5 +13,31 @@
 //= require jquery
 //= require bootstrap-sprockets
 //= require jquery_ujs
+//= require jquery.turbolinks
 //= require turbolinks
 //= require_tree .
+
+$(document).ready(function () {
+	$('.option').click(function () {
```
Comment (dieunb) [inline app/assets/javascripts/application.js]: indent (tabsize=2 spaces)

File: app/controllers/words_controller.rb [hunk 3]
```diff
@@ -3,8 +3,15 @@ class WordsController < ApplicationController
 
   def index
     @categories = Category.all
-    category_id = @categories.first.id
-    category_id = params[:category_id] unless params[:category_id].nil?
-    @words = Word.where category_id: category_id
+    category = @categories.first
+    category = @categories.find_by id: params[:category_id] unless params[:category_id].nil?
+    case params[:learned_filter]
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: refactor `case ... when` by using meta programming method `send`

**Comment by dieunb** [comment]
OK

File: app/controllers/lessons_controller.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+class LessonsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @category = Category.find params[:category_id]
+    @lesson = @category.lessons.build user_id: current_user.id
+    words = Word.filter_with_category(@category).random_words current_user
```
Comment (chuanhtuan) [inline app/controllers/lessons_controller.rb]: @category.word.random_words current_user

File: app/controllers/words_controller.rb [hunk 5]
```diff
@@ -3,8 +3,12 @@ class WordsController < ApplicationController
 
   def index
     @categories = Category.all
-    category_id = @categories.first.id
-    category_id = params[:category_id] unless params[:category_id].nil?
-    @words = Word.where category_id: category_id
+    category = @categories.first
+    category = @categories.find_by id: params[:category_id] unless params[:category_id].nil?
+    @words = if %w(learned not_learned).include? params[:learned_filter]
+      Word.filter_with_category(category).send params[:learned_filter], current_user
```
Comment (chuanhtuan) [inline app/controllers/words_controller.rb]: nt

**Pull request merged**